_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -70,12 +86,51 @@ proc _cmake_get_build_dir {} {
     return [option worksrcpath]
 }
 
-default configure.dir {[_cmake_get_build_dir]}
-
-pre-configure {
-    file mkdir ${configure.dir}
+option_proc cmake.generator cmake::handle_generator
+proc cmake::handle_generator {option action args} {
+    global cmake.generator destroot destroot.target build.cmd build.post_args
+    global depends_build destroot.post_args build.jobs
+    if {${action} eq "set"} {
+        switch -nocase -glob ${args} {
+            "*Unix Makefiles*" {
+                ui_debug "Selecting the 'Unix Makefiles' generator"
+                cmake.generator     "Unix Makefiles"
+                depends_build-delete \
+                                    port:ninja
+                build.cmd           make
+                build.post_args     VERBOSE=ON
+                destroot.target     install/fast
+                destroot.destdir    DESTDIR=${destroot}
+                # unset the DESTDIR env. variable if it has been set before
+                destroot.env-delete DESTDIR=${destroot}
+            }
+            Ninja {
+                ui_debug "Selecting the Ninja generator"
+                cmake.generator     Ninja
+                depends_build-append \
+                                    port:ninja
+                build.cmd           ninja
+                # force Ninja to use the exact number of requested build jobs
+                build.post_args     -j${build.jobs}
+                destroot.target     install
+                # ninja needs the DESTDIR argument in the environment
+                destroot.destdir    ""
+                destroot.env-append DESTDIR=${destroot}
+            }
+            default {
+                if {[file tail ${cmake::currentportgroupdir}] eq "group"} {
```

Comment:
I agree, or at least that's what I thought too. Marko has been experiencing proof of the contrary though (during `port upgrade outdated` and `port uninstall inactive`) and we haven't been able to figure out what happened exactly.

Either way, with the globbing switch I use now there is less reason to be pedantic about using quotes or not; both work (in user code) which I kind of like. One difference with what you write: when I do `ui_msg "${args}"` in the handle_proc and call `cmake.generator "Unix Makefiles"`, I see `{{Unix Makefiles}}` printed on the terminal.
This probably means that the `join` statement in the final treatment remains a good idea.

The globbing match also means we're actually supporting all current CMake generators that just generate some additional files to make it possible to open the configured project in IDEs like CodeBlocks or Eclipse.